File: testapp1202_ui/src/services/SeoStrategyService.js
import http from "../http-common"; 

class SeoStrategyService {
  getAllSeoStrategys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/seoStrategy/seoStrategys`, searchDTO);
  }

  get(seoStrategyId) {
    return this.getRequest(`/seoStrategy/${seoStrategyId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/seoStrategy?field=${matchData}`, null);
  }

  addSeoStrategy(data) {
    return http.post("/seoStrategy/addSeoStrategy", data);
  }

  update(data) {
  	return http.post("/seoStrategy/updateSeoStrategy", data);
  }
  
  uploadImage(data,seoStrategyId) {
  	return http.postForm("/seoStrategy/uploadImage/"+seoStrategyId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SeoStrategyService();

File: src/main/java/com/testapp1202/service/SeoStrategyService.java
package com.testapp1202.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.testapp1202.domain.SeoStrategy;
import com.testapp1202.dto.SeoStrategyDTO;
import com.testapp1202.dto.SeoStrategySearchDTO;
import com.testapp1202.dto.SeoStrategyPageDTO;
import com.testapp1202.dto.SeoStrategyConvertCriteriaDTO;
import com.testapp1202.service.GenericService;
import com.testapp1202.dto.common.RequestDTO;
import com.testapp1202.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SeoStrategyService extends GenericService<SeoStrategy, Integer> {

	List<SeoStrategy> findAll();

	ResultDTO addSeoStrategy(SeoStrategyDTO seoStrategyDTO, RequestDTO requestDTO);

	ResultDTO updateSeoStrategy(SeoStrategyDTO seoStrategyDTO, RequestDTO requestDTO);

    Page<SeoStrategy> getAllSeoStrategys(Pageable pageable);

    Page<SeoStrategy> getAllSeoStrategys(Specification<SeoStrategy> spec, Pageable pageable);

	ResponseEntity<SeoStrategyPageDTO> getSeoStrategys(SeoStrategySearchDTO seoStrategySearchDTO);
	
	List<SeoStrategyDTO> convertSeoStrategysToSeoStrategyDTOs(List<SeoStrategy> seoStrategys, SeoStrategyConvertCriteriaDTO convertCriteria);

	SeoStrategyDTO getSeoStrategyDTOById(Integer seoStrategyId);







}





